libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1073,7 +1104,11 @@ string ABIFunctions::abiEncodingFunctionStruct(
 					break;
 				}
 				case DataLocation::CallData:
-					solUnimplementedAssert(false, "Encoding struct from calldata is not yet supported.");
+				{
+					string sourceOffset = toCompactHexWithPrefix(_from.calldataOffsetOfMember(member.name));
+					members.back()["retrieveValue"] = calldataAccessFunction(*memberTypeFrom) + "(value, add(value, " + sourceOffset + "))";
```

Comment:
What if `calldataAccessFunction` returns multiple values?

Reply:
Then the ``retrieveValues`` call is assigned to the ``memberValues`` variables, which should be multiple names exactly in those cases as well.